backend/server/routers/programs.py

```diff
@@ -211,4 +212,21 @@ def get_structure(
             add_subgroup_container(structure, "General", container, [])
 
     
-    return {"structure": structure}
\ No newline at end of file
+    return {"structure": structure}
+
+@router.get(
+    "/getGenEds/{programCode}",
+    responses={
+        400: {
```

Comment:
There should be a response model here. You can use Courses if you take Gabriel's advice.